gobblin-metastore/src/main/java/org/apache/gobblin/metastore/MysqlJobStatusStateStore.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.gobblin.metastore;
+
+import java.io.IOException;
+import java.util.List;
+
+import javax.sql.DataSource;
+
+import org.apache.gobblin.configuration.State;
+
+
+public class MysqlJobStatusStateStore<T extends State> extends MysqlStateStore {
```

Comment:
Do we need this class? Can't the logic of appending the suffix "%" can be moved into MySqlJobStatusRetriever, so that this class can be removed?

Reply:
Had to write this just to override one method. I did not want to change the current MysqlStateStore...
In future, we may need more specialized sql queries; that's another reason I did not change MysqlStateStore